tempto-core/src/main/java/com/teradata/tempto/internal/fulfillment/table/cassandra/CassandraTableManager.java

```diff
@@ -0,0 +1,209 @@
+
+            // TODO: check if data is up-to-date
+
+            insertData(tableName, dataSource);
+        }
+
+        return new CassandraTableInstance(tableName, tableDefinition);
+    }
+
+    private void insertData(TableName tableName, RelationalDataSource dataSource)
+    {
+        checkArgument(tableName.getSchema().isPresent(), "Cassandra does not like you when you don't have keyspace defined");
+
+        Iterator<List<Object>> dataRows = dataSource.getDataRows();
+        if (!dataRows.hasNext()) {
+            return;
+        }
+
+        List<String> columnNames = queryExecutor.getColumnNames(tableName.getSchema().get(), tableName.getInternalNameInDatabase());
+
+        try {
+            CassandraBatchLoader loader = new CassandraBatchLoader(queryExecutor.getSession(), tableName.getNameInDatabase(), columnNames);
+            for (List<List<Object>> batch : partitionBy(dataRows, BATCH_SIZE)) {
+                loader.load(batch);
+            }
+        }
+        catch (SQLException e) {
+            throw new RuntimeException(e);
+        }
+    }
+
+    private static Iterable<List<List<Object>>> partitionBy(Iterator<List<Object>> dataRows, int partitionSize)
+    {
+        return () -> new Iterator<List<List<Object>>>()
+        {
+            @Override
+            public boolean hasNext()
+            {
+                return dataRows.hasNext();
+            }
+
+            @Override
+            public List<List<Object>> next()
+            {
+                List<List<Object>> batch = new ArrayList<>();
+                while (dataRows.hasNext() && batch.size() < partitionSize) {
+                    batch.add(dataRows.next());
+                }
+                return batch;
+            }
+        };
+    }
+
+    @Override
+    public TableInstance<RelationalTableDefinition> createMutable(RelationalTableDefinition tableDefinition, MutableTableRequirement.State state, TableHandle tableHandle)
+    {
+        TableName tableName = createMutableTableName(tableHandle);
+        checkArgument(tableName.getSchema().isPresent(), "Cassandra does not like you when you don't have keyspace defined");
+
+        CassandraTableInstance tableInstance = new CassandraTableInstance(tableName, tableDefinition);
+        if (state == PREPARED) {
+            return tableInstance;
+        }
+
+        queryExecutor.executeQuery(tableDefinition.getCreateTableDDL(tableName.getNameInDatabase()));
+
+        if (state == CREATED) {
+            return tableInstance;
+        }
+
+        assert state == LOADED;
+
+        RelationalDataSource dataSource = tableDefinition.getDataSource();
+        insertData(tableName, dataSource);
+        return tableInstance;
+    }
+
+    @Override
+    public void dropTable(TableName tableName)
+    {
+        queryExecutor.executeQuery("DROP TABLE " + tableName.getNameInDatabase());
+    }
+
+    public void dropTable(String tableName)
```

Comment:
One `dropTable` (implementation of the interface method) is public, so I think for consistency it would be good to have this public as well. I can change that if you want though.